prediction_market_agent/agents/safe_guard_agent/guards/llm.py

```diff
@@ -82,48 +89,179 @@ def validate_safe_transaction_llm(
 
 
 def format_transaction(tx: DetailedTransactionResponse) -> str:
+    formatted = (
+        f"Transaction ID: {tx.txId} | "
+        + (
+            f"Executed at: {DatetimeUTC.to_datetime_utc(tx.executedAt)} | "
+            if tx.executedAt
+            else ""
+        )
+        + format_tx_data(tx.txData)
+        if tx.txData
+        else ""
+    )
+    if tx.detailedExecutionInfo is not None:
+        formatted += format_detailed_execution_info(tx.detailedExecutionInfo)
+
     tx_info = tx.txInfo
 
-    sender_value = (
-        tx_info.sender.value
-        if isinstance(tx_info, TransferTxInfo)
-        else tx_info.owner
-        if isinstance(tx_info, SwapOrderTxInfo)
-        else "N/A"
+    if isinstance(tx_info, CreationTxInfo):
+        formatted += format_creation_info(tx_info)
+
+    elif isinstance(tx_info, SettingsChangeTxInfo):
+        formatted += format_settings_change_info(tx_info)
+
+    elif isinstance(tx_info, TransferTxInfo):
+        formatted += format_transfer_info(tx_info)
+
+    elif isinstance(tx_info, SwapOrderTxInfo):
+        formatted += format_swap_order_info(tx_info)
+
+    elif isinstance(tx_info, CustomTxInfo):
+        formatted += format_custom_info(tx_info)
```

Comment:
Would it be possible to add formatting to the classes themselves (e.g. `CreationTxInfo.format()` or similar) so that we can have just one call here and no type assertions?

Reply:
Sure looks nice. Refactored!